mage_ai/data_preparation/models/file.py

```diff
@@ -54,10 +43,45 @@ def delete(self):
         os.remove(self.file_path)
 
     def to_dict(self, include_content=False):
-        data = dict(
-            name=self.filename,
-            path=os.path.join(self.dir_path, self.filename)
-        )
+        data = dict(name=self.filename, path=os.path.join(self.dir_path, self.filename))
         if include_content:
             data['content'] = self.content()
         return data
+
+
+def traverse(directory: str) -> Dict:
+    """
+    Traverses directory tree and returns the underlying file structure
+
+    Args:
+        directory (str): Top-level directory to generate tree from
+
+    Returns:
+        Dict: Directory tree payload representation
+    """
+    name = os.path.splitext(os.path.basename(directory))[0]
+    disabled = name[0] == '.' or name in INACCESSIBLE_DIRS
```

Comment:
There are some duplicate code here.
You can define `_traverse` method as
```
def _traverse(name, path, disabled=True, depth=1) -> Dict:
    ....
```
Then call `_traverse(repo_name, repo_path, False, 0)`

Reply:
Fixed, although function has a lot of parameters now